azext_iot/operations/hub.py

```diff
@@ -285,31 +286,18 @@ def iot_device_update(
     target = discovery.get_target(
         hub_name=hub_name, resource_group_name=resource_group_name, login=login
     )
-    resolver = SdkResolver(target=target)
-    service_sdk = resolver.get_sdk(SdkType.service_sdk)
 
-    try:
-        auth, pk, sk = _parse_auth(parameters)
-        updated_device = _assemble_device(
-            parameters['deviceId'],
-            auth, parameters['capabilities']['iotEdge'],
-            pk,
-            sk,
-            parameters['status'].lower(),
-            parameters.get('statusReason')
-        )
-        etag = parameters.get("etag", None)
-        if etag:
-            headers = {}
-            headers["If-Match"] = '"{}"'.format(etag)
-            return service_sdk.registry_manager.create_or_update_device(
-                id=device_id, device=updated_device, custom_headers=headers
-            )
-        raise LookupError("device etag not found.")
-    except CloudError as e:
-        raise CLIError(unpack_msrest_error(e))
-    except LookupError as err:
-        raise CLIError(err)
+    auth, pk, sk = _parse_auth(parameters)
+    updated_device = _assemble_device(
+        parameters['deviceId'],
+        auth, parameters['capabilities']['iotEdge'],
+        pk,
+        sk,
+        parameters['status'].lower(),
+        parameters.get('statusReason')
+    )
+    updated_device.etag = parameters.get("etag", None)
```

Comment:
I think we can make the default etag "*" here as in `parameters.get("etag", "*")`

Reply:
Done